omc3/hole_in_one.py

```diff
@@ -497,11 +510,18 @@ def harpy_params():
                               "is up to 2 ** output_bits (maximal in case full spectra is output). "
                               "There is one pair (with maximal amplitude of complex coefficient) "
                               "per interval of size 2 ** (- output_bits - 1).")
+    params.add_parameter(name="resonances", type=int, default=HARPY_DEFAULTS["resonances"],
+                        help="Maximum magnet order of resonance lines to calculate.")
     return params
 
 
 def _optics_entrypoint(params):
-    return EntryPoint(optics_params(), strict=False).parse(params)
+    options, rest = EntryPoint(optics_params(), strict=False).parse(params)
+    
+    if options.rdt_magnet_order < 2 and 'rdt' in options.nonlinear:
+        raise AttributeError("The minimum magnet order for RDT calculation is 2.")
```

Comment:
shouldn't those also check the upper bound?

Reply:
Voilà